File: Leetcode/Longest Chunked Palindrome Decomposition/Longest Chunked Palindrome Decomposition.py
# A program to find longest chunked palindrome decomposition

class Solution(object):
    def longestDecomposition(self, text):
        lens = len(text)
        left = 0
        right = lens - 1
        count = 0
#  Temp is the length of each case        
        temp = 0
        while left < right:
# When there are no equal strings on the left and right sides of the entire string, as in Example 2, there is no palindrome at all.           
            if left + temp > right - temp:
                    count += 1
                    break
            if text[left: left + temp + 1] == text[right - temp: right + 1]:
                
                if left == right - temp:
                    count += 1
                    break
# when there is only tpe in the middle of the scan, although left <right, it is forced to exit
                else:
                    count += 2
                    left = left + temp + 1
                    right = right - temp - 1
                    temp = 0
            else:
                temp += 1

# Returning result after verification
        if left == right:
            count += 1
        return count

File: Leetcode/Longest Chunked Palindrome Decomposition/Longest Chunked Palindrome Decomposition.java
class Solution {
    public int longestDecomposition(String text) {
        int ans = 0;
        //Finding the length of given string
        int n = text.length();
        //Initialising two empty strings
        String s = "", t = "";
        //Looping through characters
        for (int i = 0; i < n; ++i) {
            //Adding characters from beginning and end to both strings respectively
            s = s + text.charAt(i);
            t = text.charAt(n - i - 1) + t;
            //If strings are equal, increasing count and procceding again with empty strings
            if (s.equals(t)) {
                ++ans;
                s = "";
                t = "";
            }
        }
        //Returning ans
        return ans;
    }
}
